### ISSUE TITLE
Allow to override Endpoint url method to handle objects in query params

### ISSUE THREAD

**Issue by uxxman** [reactive/data-client#2917]
**Is your feature request related to a problem? Please describe.**
Currently, this lib doesn't handle using deeply nested objects as query params. We need a way to allow the users to handle this on their own. As discussed here https://github.com/reactive/data-client/issues/348

**Comment by ntucker** [comment]
On holidays ATM; but I will look at this end of week. Should be pretty easy. Thanks for the report!

**Comment by uxxman** [comment]
no worries 😉 have a great week

**Comment by ntucker** [comment]
[Example implementation using qs library for search parameter serialization](https://dataclient.io/rest/api/RestEndpoint#using-qs-library)

```ts
import { RestEndpoint, RestGenerics } from '@data-client/rest';
import qs from 'qs';

class QSEndpoint<O extends RestGenerics = any> extends RestEndpoint<O> {
  searchToString(searchParams) {
    return qs.stringify(searchParams);
  }
}
```

Released in [0.10](https://github.com/reactive/data-client/releases/tag/%40data-client%2Frest%400.10.0)
